src/test/test-condition.c

```diff
@@ -125,6 +127,47 @@ static void test_condition_test_path(void) {
         condition_free(condition);
 }
 
+static void test_condition_test_control_group_controller(void) {
+        Condition *condition;
+        _cleanup_free_ const char *controller = NULL;
+        _cleanup_set_free_free_ Set *controllers = NULL;
+        int r;
+
+        condition = condition_new(CONDITION_CONTROL_GROUP_CONTROLLER, "thisisnotarealcontroller", false, false);
+        assert_se(condition);
+        assert_se(!condition_test(condition));
+        condition_free(condition);
+
+        condition = condition_new(CONDITION_CONTROL_GROUP_CONTROLLER, "thisisnotarealcontroller", false, true);
+        assert_se(condition);
+        assert_se(condition_test(condition));
+        condition_free(condition);
+
+
+        r = cg_kernel_controllers(&controllers);
+        if (r < 0) {
+                log_notice("skipping positive test, failed to get controllers");
+        } else {
+                controller = set_steal_first(controllers);
+
+                if (controller) {
+                        log_info("chosen controller is '%s'", controller);
```

Comment:
cg_kernel_controllers() returns all controllers, even those we don't support, and named hierarchies, but our condition doesn't cover those (and shouldn't I think), hence you might want to skip over controllers here for which cgroup_controller_from_string() returns negative (i.e. the ones we don't recognize)